coordinape/coordinape
# Recover claims

## Motivation and Context

https://www.notion.so/coordinape/Cron-job-for-ensuring-claim-Tx-s-are-recovered-if-lost-1f0cb54f958d443481f4935ad44d934a

## Description

- Wire up the frontend claim flow to store a record in `pending_vault_transactions`, then remove it after the transaction completes, when the full record is saved to the db
- Convert the `recoverVaults` cron job to `recoverTransactions` and modify it to also handle claim transactions

## Test and Deployment Plan

Modified unit test to cover claims as well as vault creation

Manual testing:
- submit a transaction to claim allocations in the UI, then once you see "Sending transaction...", close your browser
- When you return to the browser immediately after, there should be no record of your claim; but if you return again after 7 minutes, it should be shown in the claim history

## Other changes

Created a new directory, `api-test`, for this test, because it imports code from `src`, and doing that from within `api` breaks `yarn typecheck`
